### ISSUE TITLE
PGI Fortran compiler not happy with master and v4.0.x

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#6411]
Looks like commit 2c71c2788 broke the Fortran interfaces when using the PGI compiler (at least 17 and 18).  The comments in the commit seem to indicate that there was a notion that Fortran compilers only needed to use module files in cases where there were no routines defined in the module.  At least for PGI this isn't the case.

With PGI 17 and 18, if one uses master or v4.0.x and tries to make the examples in the examples folder, the make fails with 

```
mpicc -g  hello_c.c  -o hello_c
mpicc -g  ring_c.c  -o ring_c
mpicc -g  connectivity_c.c  -o connectivity_c
mpicc -g  spc_example.c  -o spc_example
make[1]: Entering directory `/yellow/usr/projects/hpctools/hpp/ompi/examples'
make[2]: Entering directory `/yellow/usr/projects/hpctools/hpp/ompi/examples'
mpifort -g  hello_mpifh.f  -o hello_mpifh
/usr/projects/hpctools/hpp/ompi/install_pgi18_m/lib/libmpi_usempif08.so: undefined reference to `ompi_mpifh_bindings_'
make[2]: *** [hello_mpifh] Error 2
make[2]: Leaving directory `/yellow/usr/projects/hpctools/hpp/ompi/examples'
make[1]: *** [mpi] Error 2
make[1]: Leaving directory `/yellow/usr/projects/hpctools/hpp/ompi/examples'
make: *** [all] Error 2
```

**Comment by jsquyres** [comment]
@hppritcha Does this also happen in OMPI 3.1.x and/or 3.0.x?

**Comment by hppritcha** [comment]
no it does not.  commit 2c71c27 only went in to master prior to fork of 4.0.x

**Comment by hppritcha** [comment]
Fixed in master and 4.0.x closing
